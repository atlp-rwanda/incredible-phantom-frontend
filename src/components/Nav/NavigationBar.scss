@import '../../styles/_fonts.scss';
@import '../../styles/_variables.scss';

@mixin flexCenter($direction) {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: $direction;
}

.Nav-Bar {
  width: 95%;
  height: 10vh;
  margin: auto;
  @include flexCenter(row);
  .logo {
    flex: 1;
    font-family: $headerFont;
    font-size: 2rem;
    font-weight: bolder;
  }
  .navigation {
    flex: 5;
    text-align: right;
    flex-wrap: nowrap;
    .list {
      list-style: none;
      margin: 0px 40px 0px 40px;
      font-size: 18px;
      font-family: $normalFont;
      font-weight: bolder;
      transition: 0.2s;
      text-decoration: none;
      color: inherit;
      cursor: pointer;
      button {
        @include phantomButton();
        font-size: 18px;
      }
      &:hover {
        color: $phantomHover;
      }
    }
  }
}

.resp-nav {
  position: relative;
  @include flexCenter(row);
  width: 100%;
  .logo {
    flex: 1;
    text-align: center;
    font-family: $headerFont;
    font-weight: bolder;
  }
  .resp-navigation {
    @include flexCenter(row);
    padding: 10px 0px 10px 0px;
    flex: 3;
    flex-flow: row nowrap;
    .options {
      .dropbtn {
        background-color: transparent;
        &:hover {
          background-color: transparent;
        }
      }
      button {
        margin-left: 10px;
        @include phantomButton();
        font-size: 1.1rem;
      }
    }
    .burger {
      z-index: 20;
      width: 20%;
      margin-left: 30px;
      div {
        width: 30px;
        border: 2px solid $phantomGreen;
        margin-bottom: 5px;
        transform-origin: 4px;
        transition: all 0.2s linear;
      }
    }
  }
  .links {
    position: absolute;
    background-color: #1b4332;
    left: 0;
    top: 0;
    width: 50%;
    height: 100vh;
    color: white;
    z-index: 19;
    padding-top: 40px;
    text-align: center;
    transition: 0.3s ease-in-out;
    transform: translateX(-1000px);
    a {
      list-style: none;
      display: block;
      width: 80%;
      margin: auto;
      padding: 20px 0px 20px 0px;
      border-bottom: 1px solid $phantomLight;
      font-family: $normalFont;
      text-decoration: none;
      color: inherit;
    }
    a:nth-child(4) {
      border-bottom: none;
    }
  }
  display: none;
}

@media only screen and (max-width: 916px) {
  .Nav-Bar {
    display: none;
  }
  .resp-nav {
    display: flex;
    .resp-navigation {
      @include flexCenter(row);
      padding: 10px 0px 10px 0px;
      flex: 2;
      flex-flow: row nowrap;
      .options {
        .dropbtn {
          background-color: transparent;
          &:hover {
            background-color: transparent;
          }
        }
        button {
          margin-left: 5px;
          @include phantomButton();
          font-size: 4vw;
        }
      }
      .burger {
        z-index: 20;
        width: 20%;
        margin-left: 30px;
        div {
          width: 30px;
          border: 2px solid $phantomGreen;
          margin-bottom: 5px;
          transform-origin: 4px;
          transition: all 0.3s linear;
        }
      }
    }
  }
}
