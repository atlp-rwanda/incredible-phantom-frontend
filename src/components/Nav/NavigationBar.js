import React from 'react';
import { Link } from 'react-router-dom';
import { LangSelection } from '../LangSelection/LangSelection';
import { Responsive } from '../Nav/Responsive';
import { Trans } from 'react-i18next';

export const NavigationBar = ({ home, about, contact }) => {
  return (
    <div>
      <div className="Nav-Bar">
        <div className="logo">PHANTOM</div>
        <div className="navigation">
          <LangSelection />
          <Link className="list" style={home} id="first-child" to="/">
            <Trans i18nKey="nav.home">Home</Trans>
          </Link>
          <Link className="list" style={about} id="first-child" to="/about">
            <Trans i18nKey="nav.about">About</Trans>
          </Link>
          <Link className="list" to="/terms">
            <Trans i18nKey="nav.terms">Terms of Services</Trans>
          </Link>
          <Link className="list" style={contact} to="/contact">
            <Trans i18nKey="nav.contact">Contact Us</Trans>
          </Link>
          <a className="list" style={{ margin: '0px' }}>
            <button><Trans i18nKey="nav.signup">Sign Up</Trans></button>
          </a>
        </div>
      </div>
      <Responsive />
    </div>
  );
};
