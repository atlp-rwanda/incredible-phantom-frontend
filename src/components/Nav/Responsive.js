import React from 'react';
import { Link } from 'react-router-dom';
import { LangSelection } from '../LangSelection/LangSelection';
import { useSelector, useDispatch } from 'react-redux';
import { Switch } from '../../redux/actionCreators/toggleNavAction';
import { useTranslation } from 'react-i18next';
import './NavigationBar.scss';

export const Responsive = () => {
  const toggle = useSelector((state) => state.navReducer);
  const dispatch = useDispatch();
  const { t } = useTranslation();
  return (
    <div className="resp-nav">
      <div className="logo">PHANTOM</div>
      <div className="resp-navigation">
        <div className="options">
          <LangSelection />
          <button> {t('nav.signup')}</button>
        </div>
        <div onClick={() => dispatch(Switch())} className="burger">
          <div
            style={{
              transform: toggle ? 'rotate(45deg)' : 'rotate(0)',
            }}
          />
          <div
            style={{
              transform: toggle ? 'translateX(100%)' : 'translateX(0)',
              opacity: toggle ? 0 : 1,
            }}
          />
          <div
            style={{
              transform: toggle ? 'rotate(-45deg)' : 'rotate(0)',
            }}
          />
        </div>
      </div>
      <div
        className="links"
        style={{
          transform: toggle ? 'translateX(0px)' : 'translateX(-100%)',
        }}
      >
        <Link onClick={() => dispatch(Switch())} to="/" className="link">
          {t('nav.home')}
        </Link>
        <Link onClick={() => dispatch(Switch())} to="/about" className="link">
          {t('nav.about')}
        </Link>
        <Link onClick={() => dispatch(Switch())} to="/terms" className="link">
          {t('nav.terms')}
        </Link>
        <Link onClick={() => dispatch(Switch())} to="contact" className="link">
          {t('nav.contact')}
        </Link>
      </div>
    </div>
  );
};
