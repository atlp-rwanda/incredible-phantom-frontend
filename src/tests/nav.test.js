import React from 'react';
import { shallow } from 'enzyme';
import { NavigationBar } from '../components/Nav/NavigationBar';
import { Responsive } from '../components/Nav/Responsive';
import { Provider } from 'react-redux';
import store from '../redux/reducers/index';

describe('Navigation Bar', () => {
  it('should return the Navigation Bar', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <NavigationBar />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should return the Navigation Bar responsive', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <Responsive />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
