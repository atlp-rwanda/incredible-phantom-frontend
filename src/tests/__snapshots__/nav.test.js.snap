// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Bar should return the Navigation Bar 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <NavigationBar />
</ContextProvider>
`;

exports[`Navigation Bar should return the Navigation Bar responsive 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Responsive />
</ContextProvider>
`;
